test/spmd/test_tracing.py

```diff
@@ -410,5 +414,65 @@ def f(x, y):
         self._test_expand(xd, yd, f, mesh, out_placement)
 
 
+class TraceModuleTest(DistTensorTestBase):
+    @property
+    def world_size(self):
+        return 2
+
+    def _test_trace_replicate(self, model, x):
+        # if x.device.type == "cuda":
+        ddp = DDP(deepcopy(model))
+        spmd = SPMD(
+            deepcopy(model),
+            schema=Schema(
+                mesh=DeviceMesh(
+                    self.device_type, torch.arange(self.world_size)
+                ),
+                placements=[Replicate()],
+            ),
+        )
+
+        ddp(x).sum().backward()
+        spmd(x).sum().backward()
+        for p1, p2 in zip(ddp.parameters(), spmd.parameters()):
+            # DDP divides gradients by world size to compute average, but
+            # _Partial tensor shouldn't do that automatically. Hence explicitly
```

Comment:
_partial don't need to do this divide because it's already working with batched sharded input?

Reply:
Could you please elaborate? I thought we need this because partial -> replicate triggers allreduce sum, while DDP does allreduce sum and then divide by world size. 

Whether we should do the division or not depends on the loss function. DDP chooses to apply the division because most loss function is calculating averaged per-sample loss. So dividing by world size will produce similar result as local training for the majority of use cases. 

This test is using `sum()` as loss function, so in theory DDP shouldn't divide it by world size, but however that division is hard-coded in DDP source code.